packages/@aws-cdk/aws-cognito/README.md

```diff
@@ -53,6 +66,265 @@ This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aw
   - [Resource Servers](#resource-servers)
   - [Domains](#domains)
 
+## Identity Pools
+
+Identity pools provide temporary AWS credentials for users who are guests (unauthenticated) and for users who have been authenticated and received a token. An identity pool is a store of user identity data specific to your account.
+
+Identity pools can be used in conjunction with Cognito User Pools or by accessing external federated identity providers directly.
+
+
+### Authenticated and Unauthenticated Identities
+
+Identity pools define two types of identities: authenticated and unauthenticated. Every identity in your identity pool is either authenticated or unauthenticated. Authenticated identities belong to users who are authenticated by a public login provider (Amazon Cognito user pools, Login with Amazon, Sign in with Apple, Facebook, Google, SAML, or any OpenID Connect Providers) or a developer provider (your own backend authentication process). Unauthenticated identities typically belong to guest users.
+
+A basic Identity Pool with minimal configuration consists of two roles: one to apply to authenticated identities and one to apply to unauthenticated identities:
+
+```ts
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  authenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-authenticated-role'),
+  unauthenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-unauthenticated-role'),
```

Comment:
That's a fair point and I thought about that a lot, and the unauthenticated role could potentially have a default role that rejects all incoming traffic, but the authenticated role is too specialized to have a default version. Since that's the default role applied to authenticated users, it has to have permissions specific to what is necessary for the service, and those could include almost anything.

I suppose a default authenticated role with a policy that allows for obtaining the credentials themselves could be applied for the authenticated role, and then allow the user to add policy statements to that role. Do you think that would be a better approach?

Reply:
> and then allow the user to add policy statements to that role. Do you think that would be a better approach?

I think so, yes. Since the goal of identity pool is to provide access to users, it can be treated the same as an IAM role.
We have a common pattern in the CDK called "grants".

The API could be modeled to be similar to `lambdaFunction.grantInvoke(identityPool)` which will grant permissions to the authenticated user. We could also do

Further questions in this line would be - do we need something similar for unauthenticated users? If so, how would the API look?
And, these two are only the default role. How does this look for other roles under role mapping?

Reply:
Yes, in Role Mapping the roles are provided by the user. Role Mapping is complicated enough already :).
I added methods that do something similar to this, I'll update them to follow the `grant` model.